src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/NewAzureVMSqlServerKeyVaultCredentialConfig.cs

```diff
@@ -0,0 +1,140 @@
+using System.Security;
+using System.Security.Permissions;
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Azure.Commands.Compute
+{
+    /// <summary>
+    /// Helper cmdlet to construct instance of SQL Credential AKV based settings class
+    /// </summary>
+    [Cmdlet(
+         VerbsCommon.New,
+         AzureVMSqlServerKeyVaultCredentialConfigNoun),
+     OutputType(
+         typeof(KeyVaultCredentialSettings))]
+    public class NewAzureVMSqlServerKeyVaultCredentialConfigCommand : PSCmdlet
+    {
+        /// <summary>
+        /// Configuration object friendly name
+        /// </summary>
+        protected const string AzureVMSqlServerKeyVaultCredentialConfigNoun = "AzureVMSqlServerKeyVaultCredentialConfig";
+
+        [Parameter(
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+           Mandatory = false,
+           Position = 1,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "Enable Key Vault Credential.")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter Enable { get; set; }
+
+        [Parameter(
+           Mandatory = false,
+           Position = 2,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "SQL Server credential name to create.")]
+        [ValidateNotNullOrEmpty]
+        public string CredentialName { get; set; }
+
+        [Parameter(
+           Mandatory = false,
+           Position = 3,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "Azure Key Vault service URL")]
+        [ValidateNotNullOrEmpty]
+        public string AzureKeyVaultUrl { get; set; }
+
+        [Parameter(
+           Mandatory = false,
+           Position = 4,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "Principal user client identifier.")]
+        [ValidateNotNullOrEmpty]
+        public string ServicePrincipalName { get; set; }
+
+        [Parameter(
+           Mandatory = false,
+           Position = 5,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "Principal user client secret.")]
+        [ValidateNotNullOrEmpty]
+        public SecureString ServicePrincipalSecret { get; set; }
+
+        /// <summary>
+        /// Initialzies a new instance of the <see cref="NewAzureVMSqlServerKeyVaultCredentialConfigCommand"/> class.
+        /// </summary>
+        public NewAzureVMSqlServerKeyVaultCredentialConfigCommand()
+        {
+        }
+
+        /// <summary>
+        /// Creates and returns <see cref="KeyVaultCredentialSettings"/> object.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            KeyVaultCredentialSettings settings = new KeyVaultCredentialSettings();
```

Comment:
We don't have enough context to address this comment. Can you please provide more details please